### DISCUSSION TITLE
Best practices for private and professional accounts

### DISCUSSION THREAD

**Discussion by derBroBro** [community/community#22335] [New to GitHub]
<p>Hello,</p>
<p>I searched for a while now but found no clear answer.<br>
Github recommends to use one account for private and for business use. For this you can use multiple mail-addresses.<br>
There is also a documentation for what you (as employee) should do if you leave. This includes also to remove the mail address.</p>
<p>My question is now if this can also be enforced by the company. Even if there is no access to repos anymore it may become a problem if someone commits suspicious code under your company domain.</p>
<p>Any hints how to handle this?</p>
<p>Malte</p>

**Comment by mpboom** [comment]
<p>I don’t exactly get your question, but can provide you with the following information:</p>
<p>You can associate as much mail addresses to your GitHub account as you want. If you go trough a verification process, you can also verify your linked mail addresses. That is a good thing, because I can make a commit using Git with your mail address right know. But since your mail address is tied to your account, the contribution is yours. You can always add or remove mail addresses to your account. A company can not enforce mail address removal from an account they don’t have the credentials of.</p>
<p>And yes, you should use a seperate account for work/private.</p>
<p>More information: <a href="https://help.github.com/articles/about-commit-email-addresses/" rel="nofollow noopener">https://help.github.com/articles/about-commit-email-addresses/</a></p>
<p>If you have any more questions, I’d be happy to hear.</p>

**Comment by derBroBro** [comment]
<p>Hi Mark,</p>
<p>thank you for your help.<br>
I was searching for a while now what is the best practise for private and business use. There are several perspectives for this.  The highest relevance is for me the documentation from github itself: <em>“You can use one account for multiple purposes, such as for personal use and business use. We do not recommend creating more than one account.”</em><br>
<a href="https://help.github.com/articles/differences-between-user-and-organization-accounts/" rel="nofollow noopener">https://help.github.com/articles/differences-between-user-and-organization-accounts/</a></p>
<p>My question was how to claim a domain (for mails) from a company perspective.</p>
<p>Let’s use this example:</p>
<ul>
<li>User A works for a company and claims the mail-address <a href="mailto:a@example.com">a@example.com</a> as one of his github addresses</li>
<li>User A leaves the company</li>
<li>User A commits explicit bad code to a popular repo</li>
<li>This may not be in interest of the company.</li>
</ul>
<p>I think this is a “edge case” and I hope never to come into this situation. Nevertheless it would be quite important to know. Because of this I’m searching for an option to “whitelist” the usage of this (mail)domain only for active colleagues.Also a second user would not help on this issue, why the question would be even more interesting.</p>
<p>Are there any hints how to work with this? Or just accept the risk?</p>
<p>Many thanks</p>
<p>Malte</p>

**Comment by mpboom** [comment, marked as answer]
<p>From a management perspective, I would do the following:</p>
<ul>
<li>Give employees a company mail address (<a href="mailto:employee@company.com">employee@company.com</a>) - you probably are already doing this</li>
<li>Instruct the employee to create a  <strong>new</strong> GitHub account with their company mail address - this address is then associated with the account</li>
<li>Add that account to your GitHub organization</li>
<li>Let the employee make their commits using that mail address - GitHub will link these commits to the user in your organization (this will be publicly visible)</li>
<li>Employee gets mad and quits</li>
<li>Kick the ex-employee out of the organization</li>
</ul>
<p>At this point, publicly, the account has no relation to your company  <strong>except </strong> the mail address. But that only indicates to users that this person  <strong>ever</strong> had access to that mail address and doesn’t prove much. I wouldn’t be bothered by it. However, if you are paranoid enough for it, you can just buy a cheap domain like  <strong><a href="http://fkdjslkdjfklsdjf.com" rel="nofollow noopener">fkdjslkdjfklsdjf.com</a></strong> and let your employees use that domain as a commit mail address.</p>
<p>As for creating multiple accounts - I don’t think I would be a big fan of letting users use their personal accounts at work, but on the other hand it isn’t that big of a deal either. Just as with the mail addresses, just make sure to remove employees that have left the company from the GitHub organization.</p>

**Comment by derBroBro** [comment]
<p>Hey Mark,</p>
<p>thanks a lot for you answer. These helps me a lot. :slight_smile:</p>
<p>Many regards</p>
<p>Malte</p>

**Comment by dijonkitchen** [comment]
> Tip: We recommend using only one personal account to manage both personal and professional repositories.

https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/merging-multiple-personal-accounts
